mainapp/models.py

```diff
@@ -342,21 +349,20 @@ def __str__(self):
 
 class Announcements(models.Model):
     dateadded = models.DateTimeField(auto_now_add=True)
-    name = models.CharField(max_length=50)
-    link = models.CharField(max_length=100)
-    district = models.CharField(
-        max_length = 15,
-        choices = districts,
-        verbose_name='Districts - ജില്ല'
-    )
-    category = models.IntegerField(
-        choices = announcement_types,
-        verbose_name='Type'
-    )
+    priority = models.CharField(
+        max_length=20,
+        choices = announcement_priorities,
+        verbose_name='Priority',
+        default='L')
+
+    description = models.TextField(blank=True)
+    image = models.ImageField(blank=True, null=True, upload_to='media')
```

Comment:
Don't use `null=True`, use just `blank=True`. `ImageField` and `FileField` are `CharField` internally.